srtcore/group.cpp

```diff
@@ -942,6 +942,7 @@ void CUDTGroup::close()
     vector<SRTSOCKET> ids;
 
     {
+        ScopedLock glob(CUDT::s_UDTUnited.m_GlobControlLock);
         ScopedLock g(m_GroupLock);
```

Comment:
Are you sure `CUDTGroup::close()` is not called with `s_UDTUnited.m_GlobControlLock` already acquired?

Reply:
No. This part is executed directly from the API functions and no locks are applied yet:
```
    if (u & SRTGROUP_MASK)
    {
        GroupKeeper k (*this, u, ERH_THROW);
        k.group->close();
        deleteGroup(k.group);
        return 0;
    }
```
And this `k.group->close()` is then called with no locks. Note that `GroupKeeper` only applies locks to access the group to mark the busy flag, but during "its section" all locks are released.